.spin/cmds.py

```diff
@@ -0,0 +1,138 @@
+import os
+import shutil
+import sys
+import argparse
+
+import click
+from spin.cmds import meson
+from spin import util
+
+
+@click.command()
+@click.argument("sphinx_target", default="html")
+@click.option(
+    "--clean", is_flag=True,
+    default=False,
+    help="Clean previously built docs before building"
+)
+@click.option(
+    "--build/--no-build",
+    "first_build",
+    default=True,
+    help="Build numpy before generating docs",
+)
+@click.option(
+    '--jobs', '-j',
+    metavar='N_JOBS',
+    default="auto",
+    help="Number of parallel build jobs"
+)
+@click.option(
+    "--install-deps/--no-install-deps",
+    default=True,
+    help="Install dependencies before building"
+)
+@click.pass_context
+def docs(ctx, sphinx_target, clean, first_build, jobs, install_deps):
+    """📖 Build Sphinx documentation
+
+    By default, SPHINXOPTS="-W", raising errors on warnings.
+    To build without raising on warnings:
+
+      SPHINXOPTS="" spin docs
+
+    To list all Sphinx targets:
+
+      spin docs targets
+
+    To build another Sphinx target:
+
+      spin docs TARGET
+
+    """
+    if sphinx_target not in ('targets', 'help'):
+        if install_deps:
+            util.run(['pip', 'install', '-q', '-r', 'doc_requirements.txt'])
+
+    meson.docs.ignore_unknown_options = True
+    del ctx.params['install_deps']
+    ctx.forward(meson.docs)
+
+
+@click.command()
+@click.argument("pytest_args", nargs=-1)
+@click.option(
+    "-m",
+    "markexpr",
+    metavar='MARKEXPR',
+    default="not slow",
+    help="Run tests with the given markers"
+)
+@click.option(
+    "-j",
+    "n_jobs",
+    metavar='N_JOBS',
+    default="auto",
+    help="Number of parallel jobs for testing; use all CPUs by default"
```

Comment:
I think running on all cores should be an opt-in, not a default option.

Reply:
Set the default to "1", and only pass the "-j" flag when changed. This also means that `pytest-xdist` is not required by default, which is probably a good thing.